### DISCUSSION TITLE
Using goes_latest for RadC data

### DISCUSSION THREAD

**Discussion by Tscan08** [blaylockbk/goes2go#20] [Q&A] [answered]
Hello, I'm trying to use the goes_latest tool for downloading RadC band 2. This tool worked well for the MCMIPC data, but seems to get stuck when I use the Rad product name: C = goes_latest(product='ABI-L1b-RadC'). Is this function intended to work for this product?

Is there a way to specify the band? It seems that the issue may relate to the number of files it's trying to download (code from the multiprocessing lib seems to be unhappy?).

I'm not particularly experienced with python, so hoping there is a simple solution I'm missing...

Thanks

**Comment by blaylockbk** [comment, marked as answer]
Hi @Tscan08, looks like you found a use case that goes2go didn't handle. I made a quick update (https://github.com/blaylockbk/goes2go/commit/1c356661bfdb70e7863cc3f64afc6065c39b361b) in the master branch to accept a "bands" argument to specify the ABI channel you want when requesting the `L1b-Rad` product. Try this,
```python
from goes2go.data import goes_latest
ds = goes_latest(product='ABI-L1b-RadC', bands=2, return_as='xarray')
```
![image](https://user-images.githubusercontent.com/6249613/166335918-dbe77eee-aed1-48bd-84af-5fab82c68381.png)

---
As you pointed out, I also see that the downloading of multiple files is unhappy. It looks like I need to update those functions. I'll try to fix some of these issues when I have some free time. Thanks for reporting this!

**Reply by Tscan08** [reply]
Thanks, this is working great! Really appreciate you sharing these tools.

**Reply by blaylockbk** [reply]
Glad it works for you. Happy to help.

**Reply by Tscan08** [reply]
Hi @blaylockbk, I tried this for the ABI-L2-DMWC product and it seemed to get stuck downloading multiple files from different bands.

```python
C = goes_latest(product='ABI-L2-DMWC', bands=2)
```

I'm looking do download the latest of the bands available to make a combined derived wind plot. I can specify bands individually, but it seems like the program is still getting stuck. Any suggestions?

Many thanks

**Reply by blaylockbk** [reply]
Thanks for finding and reporting another bug 😊 

I wasn't aware that the derived wind products were available for different channels. My original fix above only filtered bands for the L1 product!

I just pushed some rough changes in https://github.com/blaylockbk/goes2go/commit/cffb25935a51f771ec7d59a87650749ec754fef2 that allow filtering any ABI product by it's bands. Hope I didn't break anything in the meantime. There are some wrinkles that I need to iron out, but this worked for me...

```python
goes_latest(product='ABI-L2-DMWC', return_as='filelist')
```
<html>
<body>


file | product_mode | satellite | start | end | creation | product | mode_bands | mode | band
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
noaa-goes16/ABI-L2-DMWC/2022/136/16/OR_ABI-L2-... | ABI-L2-DMWC-M6C02 | G16 | 2022-05-16 16:16:17.100 | 2022-05-16 16:18:54.400 | 2022-05-16 16:27:57.500 | ABI-L2-DMWC | M6C02 | 6 | 2
noaa-goes16/ABI-L2-DMWC/2022/136/16/OR_ABI-L2-... | ABI-L2-DMWC-M6C07 | G16 | 2022-05-16 16:16:17.100 | 2022-05-16 16:18:55.500 | 2022-05-16 16:27:49.700 | ABI-L2-DMWC | M6C07 | 6 | 7
noaa-goes16/ABI-L2-DMWC/2022/136/16/OR_ABI-L2-... | ABI-L2-DMWC-M6C08 | G16 | 2022-05-16 16:16:17.100 | 2022-05-16 16:18:54.400 | 2022-05-16 16:27:52.500 | ABI-L2-DMWC | M6C08 | 6 | 8
noaa-goes16/ABI-L2-DMWC/2022/136/16/OR_ABI-L2-... | ABI-L2-DMWC-M6C14 | G16 | 2022-05-16 16:16:17.100 | 2022-05-16 16:18:54.400 | 2022-05-16 16:27:58.600 | ABI-L2-DMWC | M6C14 | 6 | 14


</body>
</html>

```python
goes_latest(product='ABI-L2-DMWC', return_as='filelist', bands=2)
df
```
<html>
<body>


file | product_mode | satellite | start | end | creation | product | mode_bands | mode | band
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
noaa-goes16/ABI-L2-DMWC/2022/136/16/OR_ABI-L2-... | ABI-L2-DMWC-M6C02 | G16 | 2022-05-16 16:16:17.100 | 2022-05-16 16:18:54.400 | 2022-05-16 16:27:57.500 | ABI-L2-DMWC | M6C02 | 6 | 2


</body>
</html>

Behind the scenes, that dataframe is used in the function `goes2go.data._download`. I admit, this could use some attention to make it more universal.
